geode-core/src/jmh/java/org/apache/geode/internal/cache/tier/sockets/command/ClientCachePutBench.java

```diff
@@ -0,0 +1,199 @@
+      File destServerXml = copyXmlToServerDirectory();
+
+      this.serverPort = getRandomAvailablePort(SOCKET);
+
+      List<String> command = new ArrayList<>();
+      command.add(new File(new File(getProperty("java.home"), "bin"), "java").getCanonicalPath());
+      command.add("-D" + GEMFIRE_PREFIX + CACHE_XML_FILE + "=" + destServerXml.getAbsolutePath());
+      command.add("-D" + GEMFIRE_PREFIX + MCAST_PORT + "=0");
+      command.add("-D" + GEMFIRE_PREFIX + LOCATORS + "=");
+      command.add("-D" + GEMFIRE_PREFIX + LOG_LEVEL + "=warn");
+      command.add("-cp");
+      command.add(getProperty("java.class.path"));
+      command.add(ServerLauncher.class.getName());
+      command.add(ServerLauncher.Command.START.getName());
+      command.add("server1");
+      command.add("--server-port=" + this.serverPort);
+
+      System.out.println("[ClientCachePutBench] Launching server with command: " + command);
+
+      this.process = new ProcessBuilder(command).directory(this.serverDirectory).start();
+    }
+
+    private File copyXmlToServerDirectory() throws IOException {
+      URL srcServerXml = getClass().getResource(SERVER_XML_NAME);
+      assertThat(srcServerXml).isNotNull();
+      File destServerXml = new File(this.serverDirectory, SERVER_XML_NAME);
+      copyURLToFile(srcServerXml, destServerXml);
+      return destServerXml;
+    }
+
+    private void startProcessReaders() {
+      this.processOutReader =
+          new ProcessStreamReader.Builder(this.process).inputStream(this.process.getInputStream())
+              .inputListener((line) -> System.out.println("[ClientCachePutBench][stdout] " + line))
+              .build().start();
+      this.processErrReader =
+          new ProcessStreamReader.Builder(this.process).inputStream(this.process.getErrorStream())
+              .inputListener((line) -> System.out.println("[ClientCachePutBench][stderr] " + line))
+              .build().start();
+    }
+
+    private void stopProcessReaders() throws InterruptedException {
+      if (this.processOutReader != null) {
+        this.processOutReader.stop().join(PROCESS_READER_TIMEOUT);
+      }
+      if (this.processErrReader != null) {
+        this.processErrReader.stop().join(PROCESS_READER_TIMEOUT);
+      }
+    }
+
+    @TearDown(Level.Trial)
+    public void stopServer() throws Exception {
+      System.out.println("\n" + "[ClientCachePutBench] stopServer");
+      try {
+        this.clientCache.close(false);
+        new ServerLauncher.Builder().setWorkingDirectory(this.serverDirectory.getAbsolutePath())
+            .build().stop();
+      } finally {
+        if (this.process != null) {
+          this.process.destroyForcibly();
+        }
+        this.temporaryFolder.delete();
+      }
+    }
+  }
+
+  @Benchmark
+  public String performPutFromClient(final ClientState state) throws Exception {
+    return state.region.put(createRandomString(state), createRandomString(state));
+  }
```

Comment:
I've pushed changes that reduce the overall runtime some and also moves generation of keys and values into setup. The end result seems to be no measurable difference.